### ISSUE TITLE
RankGen Training

### ISSUE THREAD

**Issue by SSamDav** [martiansideofthemoon/rankgen#5]
Hi, do you have the rankgen training script? Or just the code that you used to compute the loss?

Thanks in advance!

**Comment by martiansideofthemoon** [comment]
hi @SSamDav, in this release we are planning to release the model checkpoints along with the inference-time code for generation. The code for training RankGen is closely based on [T5X](https://github.com/google-research/t5x). We won't be releasing this now, but perhaps at some point in the future.

**Comment by SSamDav** [comment]
Ohh I see. I just have a few question about the loss:
- Did you had some stability issues with the loss?
- Did you normalize the embedding vectors before applying the dot product?

Thanks for the quick reply! This work is very interestin!

**Comment by martiansideofthemoon** [comment]
Hi @SSamDav, no we didn't have any stability issues during training.. there was variation in the loss but it usually came right back at the correct trajectory. We used adafactor with the constant learning rate of 2e-3, the standard recipe in T5X.

No we didn't normalize embeddings since that would limit the range of the logit values. I don't think normalization is done in other recent retrievers either like REALM
